### DISCUSSION TITLE
SMS verification on User Registration

### DISCUSSION THREAD

**Discussion by congngale** [zitadel/zitadel#5954] [Q&A]
Hi all,

I am working on a startup and want to build our SSO solution, I had take a look on Keycloak and it is missing 2 requirements:

1. Not support edDSA (ed25519) at the moment. They are working on it but don't have a fixed time to release this feature.
2. Having feature to verify user registration over SMS code, which will need to build a custom SPI.

Now, I am looking for alternatives and see that Zitadel along with Ory stack are two potential replacements. 

So I would like to know more about Zitadel, does it meet my 2 requirements above? If it is missing then does it easy for me to add these features into the Zitadel?

Thanks,
Cong Nga Le

**Comment by mffap** [comment]
Yes ZITADEL provides a flow to verify a users phone number. We use Twillio as service provider, you can set your API key in the instance. You can also configure the phone number as additional or primary user name (https://zitadel.com/docs/guides/solution-scenarios/configurations#use-phone-number-to-login).

**Reply by congngale** [reply]
Thanks @mffap,

Other than Twillio, Can I use other service provider for sending SMS? And , How about the EdDSA support?

**Reply by mffap** [reply]
Twillio: no not at the moment, see related discussion here: https://github.com/zitadel/zitadel/issues/5101#issuecomment-1443561533

If you have a specific provider in mind, then you could also open a discussion (Idea), so we know how much demand there is

**Reply by mffap** [reply]
> How about the EdDSA support?

I think not supported at the moment. Same as above, best to open a discussion (idea). Thanks!
